internal/webserver/webserver.go

```diff
@@ -0,0 +1,84 @@
+package webserver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	nlogger "github.com/neutron-org/neutron-logger"
+
+	"go.uber.org/zap"
+
+	"github.com/neutron-org/neutron-query-relayer/internal/relay"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	ServerContext           = "webserver"
+	UnsuccessfulTxsResource = "/unsuccessful-txs"
+)
+
+type HandlerFunc func(w http.ResponseWriter, r *http.Request)
+
+func Run(ctx context.Context, logRegistry *nlogger.Registry, storage relay.Storage, WebServerPort int) error {
+	server := &http.Server{
+		Addr:    fmt.Sprintf(":%d", WebServerPort),
+		Handler: Router(logRegistry, storage),
+	}
+	logger := logRegistry.Get(ServerContext)
+	errch := make(chan error)
+
+	go func() {
+		if err := server.ListenAndServe(); err != nil {
+			if err != http.ErrServerClosed {
+				logger.Error("failed to serve webserver", zap.Error(err))
+			}
+			errch <- err
+		}
+	}()
+
+	select {
+	case err := <-errch:
+		return err
+	case <-ctx.Done():
+	}
+
+	logger.Info("shutting down the api webserver")
```

Comment:
do we really need this complicated overloading shutting down for such a simple webserver? it won't hurt if we remove it, but it would simplify code here and in the start.go to a simple `go webserver.Run` call. we barely care about opened and idle connections that are gracefully closed in the Shutdown method, it's a kind of internal usage readonly service

Reply:
I think we already have it, so why delete it.
Also:
Webserver will also handle more complicated actions, such as resubmit failed tx (see [task](https://github.com/neutron-org/neutron-query-relayer/pull/65/files))
This action would probably require more careful shutdown